src/common.rs

```diff
@@ -74,16 +77,16 @@ pub fn invoke<C: FromIterator<Option<I>> + 'static, I>(
                 Some(ColumnarValue::Array(a)) => {
                     if let Some(str_path_array) = a.as_any().downcast_ref::<StringArray>() {
                         let paths = str_path_array.iter().map(|opt_key| opt_key.map(JsonPath::Key));
-                        zip_apply(json_array, paths, jiter_find)
+                        zip_apply(json_array, paths, jiter_find, true)
                     } else if let Some(str_path_array) = a.as_any().downcast_ref::<LargeStringArray>() {
                         let paths = str_path_array.iter().map(|opt_key| opt_key.map(JsonPath::Key));
-                        zip_apply(json_array, paths, jiter_find)
+                        zip_apply(json_array, paths, jiter_find, true)
                     } else if let Some(int_path_array) = a.as_any().downcast_ref::<Int64Array>() {
                         let paths = int_path_array.iter().map(|opt_index| opt_index.map(Into::into));
-                        zip_apply(json_array, paths, jiter_find)
+                        zip_apply(json_array, paths, jiter_find, false)
                     } else if let Some(int_path_array) = a.as_any().downcast_ref::<UInt64Array>() {
                         let paths = int_path_array.iter().map(|opt_index| opt_index.map(Into::into));
-                        zip_apply(json_array, paths, jiter_find)
+                        zip_apply(json_array, paths, jiter_find, false)
```

Comment:
Why `false` here and `true` above? This bit of logic is melting my brain 😖

Reply:
The argument is `object_lookup`, so
* `false` means "look in the array column of the union"
* `true` means "look in the object column of the union"

The point is that the input array here is an int array, hence it can only match against an array.

Reply:
Oh I see. I feel like I'd slightly prefer to have `zip_apply` work this out from `paths.first()`, but I'll live if you keep it as-is.

Reply:
Unfortunately it can't without something funky to start iterating over the path, then chain that value to the rest of the iterator.